abci/server/socket_server.go

```diff
@@ -154,14 +161,18 @@ func (s *SocketServer) handleRequests(closeConn chan error, conn net.Conn, respo
 			if err == io.EOF {
 				closeConn <- err
 			} else {
-				closeConn <- fmt.Errorf("Error reading message: %v", err.Error())
+				closeConn <- fmt.Errorf("error reading message: %v", err)
 			}
 			return
 		}
 		s.appMtx.Lock()
 		count++
 		s.handleRequest(req, responses)
 		s.appMtx.Unlock()
+
+		if panicVal != nil {
```

Comment:
why not do it inside the defer?

Reply:
Because returning from a defer does not break the execution, this is a good catch btw - I will add a return.